### PR TITLE
Setting transaction.sampled in errors

### PR DESCRIPTION
**Pull request by eyalkoren**

### CODE REVIEW

File: apm-agent-core/src/main/java/co/elastic/apm/agent/report/IntakeV2ReportingEventHandler.java [hunk 1]
```diff
@@ -165,6 +165,9 @@ private void handleEvent(ReportingEvent event, long sequence, boolean endOfBatch
             }
             connection = startRequest();
             payloadSerializer.serializeMetaDataNdJson(metaData);
+            if (logger.isTraceEnabled()) {
+                logger.trace(payloadSerializer.toString());
```
Comment (felixbarny) [inline apm-agent-core/src/main/java/co/elastic/apm/agent/report/IntakeV2ReportingEventHandler.java]: The payload is already logged when the request ends. What's the benefit of logging the whole payload for every event?

Comment (eyalkoren) [reply apm-agent-core/src/main/java/co/elastic/apm/agent/report/IntakeV2ReportingEventHandler.java]: Getting it in JSON format, see what we actually send the server

Comment (felixbarny) [reply apm-agent-core/src/main/java/co/elastic/apm/agent/report/IntakeV2ReportingEventHandler.java]: But that's already done here: https://github.com/elastic/apm-agent-java/blob/035ba3a2ef30262d6e1983daf36d92128bd76ce2/apm-agent-core/src/main/java/co/elastic/apm/agent/report/serialize/DslJsonSerializer.java#L111

Comment (eyalkoren) [reply apm-agent-core/src/main/java/co/elastic/apm/agent/report/IntakeV2ReportingEventHandler.java]: Only if flushed, which never happens if the connection with the server is down: 
https://github.com/elastic/apm-agent-java/blob/035ba3a2ef30262d6e1983daf36d92128bd76ce2/apm-agent-core/src/main/java/co/elastic/apm/agent/report/IntakeV2ReportingEventHandler.java#L282

Comment (eyalkoren) [reply apm-agent-core/src/main/java/co/elastic/apm/agent/report/IntakeV2ReportingEventHandler.java]: Anyway, not sure the right way to go, I'll remove it anyway and we will find a way to do it properly

**Comment by eyalkoren** [comment]
related to #389

**Pull request merged**